src/Workspaces/Remote/ServiceHub/Services/DiagnosticAnalyzer/DiagnosticComputer.cs

```diff
@@ -91,7 +95,11 @@ internal class DiagnosticComputer
             catch
             {
                 // Do not re-use cached CompilationWithAnalyzers instance in presence of an exception, as the underlying analysis state might be corrupt.
-                s_compilationWithAnalyzersCache.Remove(_project);
+                lock (s_gate)
+                {
+                    s_compilationWithAnalyzersCache.SetTarget(null);
```

Comment:
Does this need to check if this is the cache entry that matches _project? Otherwise if we have two things running in parallel, we won't end up erasing some other entirely valid cache? I get functionally this is fine either way, but might help perf in some cases?

Reply:
Makes sense, updated the code.